internal/elasticsearch/client.go

```diff
@@ -87,39 +85,36 @@ type Client interface {
 	GetIndexTemplates() (map[string]estypes.GetIndexTemplate, error)
 	UpdateTemplatePrimaryShards(shardCount int32) error
 
-	SetSendRequestFn(fn FnEsSendRequest)
+	// set esclient
+	SetESClient(elasticsearch6.Client)
 }
 
-type FnEsSendRequest func(cluster, namespace string, payload *EsRequest, client k8sclient.Client)
-
 type esClient struct {
-	cluster         string
-	namespace       string
-	k8sClient       k8sclient.Client
-	fnSendEsRequest FnEsSendRequest
-}
-
-type EsRequest struct {
-	Method          string // use net/http constants https://golang.org/pkg/net/http/#pkg-constants
-	URI             string
-	RequestBody     string
-	StatusCode      int
-	RawResponseBody string
-	ResponseBody    map[string]interface{}
-	Error           error
+	cluster   string
+	namespace string
+	client    elasticsearch6.Client
+	k8sclient k8sclient.Client
 }
 
+// NewClient Getting new client
 func NewClient(cluster, namespace string, client k8sclient.Client) Client {
+	u := fmt.Sprintf("https://%s.%s.svc:9200", cluster, namespace)
+	esclient6, err := getESClient(u)
+
+	if err != nil {
+		return nil
+	}
+
 	return &esClient{
-		cluster:         cluster,
-		namespace:       namespace,
-		k8sClient:       client,
-		fnSendEsRequest: sendEsRequest,
+		cluster:   cluster,
+		namespace: namespace,
+		client:    *esclient6,
+		k8sclient: client,
 	}
 }
 
-func (ec *esClient) SetSendRequestFn(fn FnEsSendRequest) {
-	ec.fnSendEsRequest = fn
+func (ec *esClient) SetESClient(client elasticsearch6.Client) {
```

Comment:
since this is only going to be used for testing, can we make this `setESClient` instead? (changing its scope)

Reply:
Ok sure make sense.

Reply:
This is done.